src/seedu/addressbook/AddressBook.java

```diff
@@ -305,12 +308,40 @@ private static void setupDefaultFileForStorage() {
     }
 
     /**
-     * Returns true if the given file is acceptable.
-     * The file path is acceptable if it ends in '.txt'
-     * TODO: Implement a more rigorous validity checking.
+     * Returns true if the given file path is valid.
+     * A file path is valid if it has a valid parent directory as determined by {@link #hasValidFolder}
+     * and a valid file name as determined by {@link #hasValidFileName}.
      */
     private static boolean isValidFilePath(String filePath) {
-        return filePath.endsWith(".txt");
+        if (filePath == null) {
+            return false;
+        }
+        Path filePathToValidate = null;
```

Comment:
My guess is 319. Usually the comment is added immediately below the offending line.